src/main/java/org/spongepowered/api/entity/ShulkerBullet.java

```diff
@@ -24,6 +24,20 @@
  */
 package org.spongepowered.api.entity;
 
-public interface ShulkerBullet extends Entity {
+import org.spongepowered.api.entity.living.golem.Shulker;
+import org.spongepowered.api.entity.projectile.DamagingProjectile;
+import org.spongepowered.api.entity.projectile.EntityTargetingProjectile;
+import org.spongepowered.api.util.Direction;
 
+/**
+ * A {@link ShulkerBullet} is the projectile {@link Shulker}s shoot.
+ */
+public interface ShulkerBullet extends EntityTargetingProjectile, DamagingProjectile {
+
+    /**
+     * The direction this {@link ShulkerBullet} is going to.
+     *
+     * @return the direction of this bullet.
+     */
+    Direction getDirection();
```

Comment:
Can be default implemented.

Reply:
What should I put it the default implementation? There is no data for it in Entity.
And I'm wondering if it would be better to expose the Direction through the Data API instead of a getter.

Reply:
Oh, I thought you already exposed DirectionalData for the bullet as well

Reply:
Can very well expose directional data on the bullet.